pydantic/_internal/_decorators.py

```diff
@@ -218,3 +229,106 @@ def in_ipython() -> bool:
         return False
     else:  # pragma: no cover
         return True
+
+
+class OnlyValueValidator(Protocol):
+    def __call__(self, __value: Any) -> Any:  # pragma: no cover
+        ...
+
+
+class V1ValidatorWithValues(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any]) -> Any:  # pragma: no cover
+        ...
+
+
+class V1ValidatorWithValuesKwOnly(Protocol):
+    def __call__(self, __value: Any, *, values: dict[str, Any]) -> Any:  # pragma: no cover
+        ...
+
+
+class V1ValidatorWithKwargs(Protocol):
+    def __call__(self, __value: Any, **kwargs: Any) -> Any:  # pragma: no cover
+        ...
+
+
+class V1ValidatorWithKwargsAndValue(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:  # pragma: no cover
+        ...
+
+
+class V1ValidatorWithValuesAndKwargs(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:  # pragma: no cover
+        ...
+
+
+V1Validator = Union[
+    V1ValidatorWithValues, V1ValidatorWithValuesKwOnly, V1ValidatorWithKwargs, V1ValidatorWithValuesAndKwargs
+]
```

Comment:
Are we emitting any kind of deprecation warning if a V1 signature is used? We'll probably want to eliminate the big union at some point if it makes mypy errors highly verbose, just want to make sure we put something in place to let us do that. (Or decide we intend to support these signatures long term, I guess)

Reply:
(Can add the warning later if you just want to get this merged sooner and don't want the distraction — assuming you and Samuel agree on that point anyway.)

Reply:
Deprecation warnings added